src/Avalonia.Base/Media/RectangleGeometry.cs

```diff
@@ -8,6 +7,18 @@ namespace Avalonia.Media
     /// </summary>
     public class RectangleGeometry : Geometry
     {
+        /// <summary>
+        /// Defines the <see cref="RadiusX"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> RadiusXProperty =
+            AvaloniaProperty.Register<RectangleGeometry, double>(nameof(RadiusX));
+
+        /// <summary>
+        /// Defines the <see cref="RadiusY"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> RadiusYProperty =
+            AvaloniaProperty.Register<RectangleGeometry, double>(nameof(RadiusY));
```

Comment:
Note `RectangleGeometry`, `Rectangle` (and even `EllipseGeometry` each defined `RadiusX` and `RadiusY` properties independently. There is no usage of AddOwner. This follows what is done in WPF and makes sense I think due to the layering here between Controls and more core/base types like geometry.

Reply:
I understand separating Ellipse*** and Rectangle*** properties completely, but not sure about Rectangle and RectangleGeometry? 

On other hand, I don't see any use cases when it's realistically useful to reuse the same property here.

Reply:
A geometry (RectangleGeometry) is conceptually quite a bitter different than a control/shape (Rectangle). Their use cases are different and implementations are very different. They exist at different levels of the framework for different things. So to me it makes sense not to use AddOwner -- but I expected some push back so made the note.

Also, when in doubt, I follow upstream:

 - [WPF `RectangleGeometry`](https://github.com/dotnet/wpf/blob/c79cdd3ef07fc3cb8954178e5c5decfb51f237ca/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Generated/RectangleGeometry.cs#L380-L397) Note that I'm not sure why these properties are auto-generated. Perhaps that influenced the original decision.
 - [WPF `Rectangle` shape](https://github.com/dotnet/wpf/blob/c79cdd3ef07fc3cb8954178e5c5decfb51f237ca/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Shapes/Rectangle.cs#L51-L101)